go/libkb/globals.go

```diff
@@ -226,7 +228,7 @@ func (g *GlobalContext) createLoginStateLocked() {
 		g.loginState.Shutdown()
 	}
 	g.loginState = NewLoginState(g)
-	g.ActiveDevice = new(ActiveDevice)
+	g.ActiveDevice.clear(g.loginState.account)
```

Comment:
I could abandon this change and leave us at the mercy of single pointer assignment concurrency. I think I'll try instead of this to stick with reassigning it and using a single lock just for this, just a sec